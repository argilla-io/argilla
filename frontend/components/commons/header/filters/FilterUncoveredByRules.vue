<template>
  <ReCheckbox
    @change="changeUncoveredByRules"
    class="re-checkbox--dark"
    :value="filter.selected"
  >
    Only records not covered by rules
  </ReCheckbox>
</template>

<script>
import { TextClassificationDataset } from "@/models/TextClassification";
export default {
  props: {
    dataset: {
      type: TextClassificationDataset,
      required: true,
    },
    filter: {
      type: Object,
      required: true,
    },
  },
  methods: {
    async changeUncoveredByRules() {
      const rulesId = this.dataset.rules.map((r) => r.query);
      this.$emit("apply", this.filter, !this.filter.selected ? rulesId : []);
    },
  },
};
</script>

<style lang="scss" scoped>
.re-checkbox {
  flex-direction: row-reverse;
  ::v-deep .checkbox-label {
    margin-left: 1em;
  }
}
</style>
