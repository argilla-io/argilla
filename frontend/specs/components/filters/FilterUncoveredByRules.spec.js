import { mount } from "@vue/test-utils";
import FilterUncoveredByRules from "@/components/commons/header/filters/FilterUncoveredByRules";

function mountFilterUncoveredByRules() {
  return mount(FilterUncoveredByRules, {
    propsData: {
      dataset: {
        task: "TextClassification",
        rules: ["alta", "baja"],
      },
      filter: {
        selected: true,
      },
    },
  });
}

describe("FilterUncoveredByRules", () => {
  let spy = jest.spyOn(console, "error");
  afterEach(() => spy.mockReset());

  test("renders properly", () => {
    const wrapper = mountFilterUncoveredByRules();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
