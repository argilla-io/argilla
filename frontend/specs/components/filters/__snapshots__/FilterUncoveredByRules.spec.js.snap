// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterUncoveredByRules renders properly 1`] = `
<recheckbox value="true" class="re-checkbox--dark">
  Only records not covered by rules
</recheckbox>
`;
